File: Java/347.java
import java.util.*;
import javafx.util.Pair;

class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> count = new HashMap<>();
        for (int i : nums) {
            count.put(i, count.getOrDefault(i, 0) + 1);
        }
        Queue<Pair<Integer, Integer>> heap = new PriorityQueue<>((x, y) -> Integer.compare(x.getValue(), y.getValue()));
        for (Map.Entry<Integer, Integer> pair : count.entrySet()) {
            heap.add(new Pair<Integer, Integer>(pair.getKey(), pair.getValue()));
            if (heap.size() > k) {
                heap.poll();
            }
        }

        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = heap.poll().getKey();
        }
        return result;
    }
}

File: Python/347.py
from typing import List


class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        for i in nums:
            count[i] = count.get(i, 0) + 1

        freq = [[] for _ in range(len(nums) + 1)]
        for num, cnt in count.items():
            freq[cnt].append(num)

        result = []
        for i in range(len(freq) - 1, 0, -1):
            for j in freq[i]:
                result.append(j)
                if len(result) == k:
                    return result
